doc/api/tls.md

```diff
@@ -876,6 +876,30 @@ socket.on('end', () => {
 });
 ```
 
+Passing in `net.Socket` will start a new instance.
+To upgrade an existing instance of `net.Socket` to a 
+`tls.TLSSocket`, pass it to `tls.connect()` as 
+the socket option: 
+
+```js
+const { Socket } = require('net');
+const tls = require('tls');
+const sock = new Socket();
+const secureSock = tls.connect({ socket: sock }, () => {
+  console.log('The tls socket connected.');
+});
+```
+
+If using TLS as the initial default rather than `net.Socket`, 
+use only `tls.connect()` to upgrade the socket:
```

Comment:
I think "upgrade" is a bit of a misnomer as there is no initial socket to be upgraded. I'd go with just:

> If no socket is provided, this function will create a new TLS socket.

as opposed to

> If a `net.Socket` is provided, this function will upgrade that TCP socket to a TLS one.